construo/src/main/java/io/github/fourlastor/construo/foojay/FooJayClient.kt

```diff
@@ -0,0 +1,90 @@
+package io.github.fourlastor.construo.foojay
+
+import com.squareup.moshi.Moshi
+import io.github.fourlastor.construo.Target
+import io.github.fourlastor.construo.ToolchainOptions
+import okhttp3.HttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.gradle.jvm.toolchain.JvmVendorSpec
+import java.io.IOException
+
+class FooJayClient {
+
+    private val okHttpClient = OkHttpClient()
+    private val moshi = Moshi.Builder().build()
+
+    fun getPackageInfo(
```

Comment:
Nit: this function is very nest-y and very long, can it be broken down for clarity?

Reply:
[c02b2bf](https://github.com/fourlastor-alexandria/construo/pull/39/commits/c02b2bf9e44accb9d20d15623189c5e8dbf8daea) good point